github/commitstatus.go

```diff
@@ -244,15 +254,23 @@ func checkForRetry(res httpResponse, waitTime, maxSleepTime time.Duration) time.
 	// If you exceed the rate limit, the response will have a 403 status and the x-ratelimit-remaining header will be 0
 	// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
 	case res.statusCode == http.StatusForbidden && res.rateLimitRemaining == 0:
-		sleepTime := time.Until(res.rateLimitReset)
+		sleepTime := res.rateLimitReset.Sub(res.date) + jitter
 		if sleepTime > maxSleepTime {
 			return 0
 		} else {
-			return sleepTime
+			// Don't allow negative sleep times. Minimum sleep time should be equal to jitter time.
+			return max(sleepTime, jitter)
```

Comment:
if we need to sleep, the minimal sleep time will a random number between 0-30.

Reply:
I now start to think that this line is wrong. Above we just added jitter to sleepTime, here there should be no need to calculate any max, we should just return sleepTime...

Reply:
Yes, I agree with you but I wanted to ensure that no negative time can happen anymore.

So my thinking was that this line:
```
sleepTime := res.rateLimitReset.Sub(res.date) 
```
can still return anything if there is any change on the server side. So adding 30sec to something negative and larger than 30sec can still create a negative sleeptime.

It should never happend of course, and `res.rateLimitReset.Sub(res.date) ` should really always be a positive value, but I was sceptical... 

So I wanted to ensure we really can't have the negative time anymore...hence `max(sleeptime, jitter)`

I am fine with removing it, just wanted to make it clear that:
- I did this on purpose
- mostly in "panic", "paranoid" mode trying to ensure that we don't have negative sleep times anymore

Reply:
I see. The problem with that approach is that the intent was hidden, so a reader is left thinking that the code is wrong. Now that I know the intent, I can advice.

From my understanding, the scenario you are mentioning can happen only in case of a clock severely misbehaving either on the server or client side. If we start going down this slope, we will never finish with "what if" scenarios, adding complexity to the code.

For these reasons, this max should be removed, eventually replaced by a log.Warning that an unexpected negative time has been detected. This log line should be added at the first moment where we can detect the problem, when we calculate sleepTime, before adding any jitter.

Reply:
Still to do: 
- [ ] add warn log line if sleep time is negative